### ISSUE TITLE
unable to build pip package from source

### ISSUE THREAD

**Issue by patelprateek** [tensorflow/tensorflow#30629]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): source
- TensorFlow version: 1.13.2 commit
- Python version: 2.7
- Installed using virtualenv? pip? conda?: pip
- Bazel version (if compiling from source): 0.19.2
- GCC/Compiler version (if compiling from source): 4.8.5
- CUDA/cuDNN version: no
- GPU model and memory: no

**Describe the problem**
```
 File "/home/p.patel/.cache/bazel/_bazel_root/1bd8cf0b6a86ac78985c768eb3530621/execroot/org_tensorflow/bazel-out/host/bin/tensorflow/create_tensorflow.python_api_1_tf_python_api_gen_v1.runfiles/org_tensorflow/tensorflow/python/tools/api/generator/create_python_api.py", line 27, in <module>
    from tensorflow.python.tools.api.generator import doc_srcs
  File "/home/p.patel/.cache/bazel/_bazel_root/1bd8cf0b6a86ac78985c768eb3530621/execroot/org_tensorflow/bazel-out/host/bin/tensorflow/create_tensorflow.python_api_1_tf_python_api_gen_v1.runfiles/org_tensorflow/tensorflow/python/__init__.py", line 63, in <module>
    from tensorflow.python.framework.framework_lib import *  # pylint: disable=redefined-builtin
  File "/home/p.patel/.cache/bazel/_bazel_root/1bd8cf0b6a86ac78985c768eb3530621/execroot/org_tensorflow/bazel-out/host/bin/tensorflow/create_tensorflow.python_api_1_tf_python_api_gen_v1.runfiles/org_tensorflow/tensorflow/python/framework/framework_lib.py", line 52, in <module>
    from tensorflow.python.framework.importer import import_graph_def
  File "/home/p.patel/.cache/bazel/_bazel_root/1bd8cf0b6a86ac78985c768eb3530621/execroot/org_tensorflow/bazel-out/host/bin/tensorflow/create_tensorflow.python_api_1_tf_python_api_gen_v1.runfiles/org_tensorflow/tensorflow/python/framework/importer.py", line 28, in <module>
    from tensorflow.python.framework import function
  File "/home/p.patel/.cache/bazel/_bazel_root/1bd8cf0b6a86ac78985c768eb3530621/execroot/org_tensorflow/bazel-out/host/bin/tensorflow/create_tensorflow.python_api_1_tf_python_api_gen_v1.runfiles/org_tensorflow/tensorflow/python/framework/function.py", line 36, in <module>
    from tensorflow.python.ops import resource_variable_ops
  File "/home/p.patel/.cache/bazel/_bazel_root/1bd8cf0b6a86ac78985c768eb3530621/execroot/org_tensorflow/bazel-out/host/bin/tensorflow/create_tensorflow.python_api_1_tf_python_api_gen_v1.runfiles/org_tensorflow/tensorflow/python/ops/resource_variable_ops.py", line 39, in <module>
    from tensorflow.python.ops import variables
  File "/home/p.patel/.cache/bazel/_bazel_root/1bd8cf0b6a86ac78985c768eb3530621/execroot/org_tensorflow/bazel-out/host/bin/tensorflow/create_tensorflow.python_api_1_tf_python_api_gen_v1.runfiles/org_tensorflow/tensorflow/python/ops/variables.py", line 133, in <module>
    "* `ONLY_FIRST_TOWER`: Deprecated alias for `ONLY_FIRST_REPLICA`.\n  ")
AttributeError: attribute '__doc__' of 'type' objects is not writable
Target //tensorflow/tools/pip_package:build_pip_package failed to build
Use --verbose_failures to see the command lines of failed build steps.
```

**Provide the exact sequence of commands / steps that you executed before running into the problem**

bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by ravikyram** [comment]
Just to verify did you get chance to follow instructions from [TensorFlow](https://www.tensorflow.org/install/source) website .Please, let us know. Thanks!

**Comment by patelprateek** [comment]
Yes

**Comment by gunan** [comment]
@mihaimaruseac who did the release for 1.13.2
Did we see this issue in the release branch?

Also, did you make any changes to the source?

**Comment by mihaimaruseac** [comment]
I didn't change the source.

Also, the release landed after the issue was opened. I'd try a rebuild now? Can you also try a newer version of Bazel? I used 0.21.0 for the release and everything worked

**Comment by ravikyram** [comment]
@patelprateek 

Can you please confirm if the issue still persists.Thanks!

**Comment by chuanqi129** [comment]
Also have this issue too

**Comment by mihaimaruseac** [comment]
Just to confirm, @chuanqi129 did you do a fresh checkout of the branch, are using bazel 0.21.0, ran `./configure` before starting the compile?

What python and compiler are you using? What operating system?

**Comment by axa000** [comment]
hi, @mihaimaruseac 
I followed the instruction on Tensorflow 
https://www.tensorflow.org/install/source_windows
I had a hard time when doing this command

bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package
(I also found others said: change to -c opt, but not working either)
most of the error said 
FAILED: Build did NOT complete successfully(326 packages loaded, 7927 targets c
onfigured)  <== Something like this

my OS : windows server 2012 r2
Bazel version: 0.28.1, also tried so many other versions 0.21.0, 0.24.1 etc
tensorflow version: 1.13.1
python version: 3.6.8

> using bazel 0.21.0, ran ./configure before starting the compile?

Does this mean every time I change Bazel version need to start all over again?

thanks a lot!

**Comment by axa000** [comment]
@mihaimaruseac hi, I created new issue incase you don't understand what I'm talking about
[LINK](https://github.com/tensorflow/tensorflow/issues/31944#issue-484800011)

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by mihaimaruseac** [comment]
Closing this issue as original asker didn't reply at all and there is a duplicate.
